lib/rules/no-implicit-coercion.js

```diff
@@ -12,7 +12,7 @@ const astUtils = require("./utils/ast-utils");
 //------------------------------------------------------------------------------
 
 const INDEX_OF_PATTERN = /^(?:i|lastI)ndexOf$/u;
-const ALLOWABLE_OPERATORS = ["~", "!!", "+", "*"];
+const ALLOWABLE_OPERATORS = ["~", "!!", "+", "--", "-", "*"];
```

Comment:
I think this means `-(-expr)` correct? If so, I think this is misleading since this looks like the decrement operator (`--`) which does change the value. I think we need to rethink how this is represented.

Reply:
I see, it does look like that, yeah

I could use `-` for both but they are different operators (unary negation and subtraction)

Maybe `"-(-)"`?

Reply:
We could use `"- -"` with a space in between (as in `- -1`) to separate the `-` tokens. It's unfortunate that those exceptions to the rule are expressed in terms of _operators_ and not _operations_, so there is no way to differentiate among `+foo` and `"" + foo` and `foo += ""` when defining an exceptions. But since there is already `"!!"` as a combination of two operators, `"- -"` should make sense, too, I guess.

Reply:
I agree with what you said